crates/iceberg/src/spec/mod.rs

```diff
@@ -18,6 +18,10 @@
 //! Spec for Iceberg.
 
 pub mod datatypes;
+pub mod partition;
```

Comment:
How about using the following export style like:

```rust
mod partition;
pub partition::Partition;
```

So our users don't need to care about which mod this type lives in?